src/services/subscription-deferred-install-service.ts

```diff
@@ -0,0 +1,66 @@
+import { Request } from "express";
+import IORedis from "ioredis";
+import { getRedisInfo } from "config/redis-info";
+import { v4 as newUUID } from "uuid";
+
+export interface SubscriptionDeferredInstallPayload {
+	installationIdPk: number,
+	gitHubServerAppIdPk?: number,
+	gitHubInstallationId: number,
+	orgName: string
+}
+
+const redis = new IORedis(getRedisInfo("SubscriptionDeferredInstallService"));
+
+const REDIS_CLEANUP_TIMEOUT = 2 * 24 * 3600 * 1000;
+
+const toRedisKey = (requestId: string) => "sub_def_inst_serv_req}" + requestId;
+
+const toRequestId = (req: Request) => req.params["requestId"];
+
+export const extractSubscriptionDeferredInstallPayload = async (req: Request) => {
+	try {
+		const requestId = toRequestId(req);
+		if (!requestId) {
+			throw new Error("Empty request ID");
```

Comment:
/nit

This is debatable on how should the code flow works? Exception (like here) or return errorCode?

So is the consumer supposed to catch when calling this api and analyse the text? 

I understand this is spike, so this is just /nit.

Personally I slightly lean towards return an error code instead of throw, but I am not aware what's the consistent behaviour in our code yet, so happy for you to decide.

Reply:
Well, given that's "extract" (action! do it!), I think exception makes more sense, cause we are not expecting any error... "validate/verify" make more sense to return a error code. But that's my IMHO